py/dml/ctree.py

```diff
@@ -4094,7 +4094,22 @@ def decref(self):
     def read(self):
         return self.expr.read()
     def assign_to(self, dest, typ):
-        mkCopyData(self.expr.site, self.expr, dest).toc()
+        # Assigns to (partly) const-qualified targets can happen as part of
+        # initializing (partly) const-qualified session variables. To allow for
+        # such assigns while avoiding GCC errors, memcpy is used together with
+        # a cast to (void *).
+        # Since session variables are allocated on the heap, this should *not*
+        # be UB as long as the session variable hasn't been initialized
+        # previously.
+        site = self.expr.site
+        if deep_const(typ):
```

Comment:
Is the alternative (current implementation) to get an _incorrect_ initialization of partially-const extern typed session, or just that it will give GCC errors? I am fairly ok with GCC errors coming from extern-related things